File: python_solutions/max_consecutive_ones.py
"""
Leetcode: https://leetcode.com/explore/learn/card/fun-with-arrays/521/introduction/3238/
"""

from typing import List


class Solution:  # pylint: disable=too-few-public-methods
    """solution class"""

    def max_consecutive_ones(self, nums: List[int]) -> int:
        """main_method"""
        max_ones = 0
        counter = 0

        for vals in nums:
            if vals == 1:
                counter += 1
                max_ones = max(counter, max_ones)
            else:
                counter = 0

        return max_ones

File: java/src/main/java/MaxConsecutiveOnes.java
class MaxConsecutiveOnes {
  public int findMaxConsecutiveOnes(int[] nums) {
    var max = 0;
    var count = 0;

    for (var num : nums) {
      if (num == 0) {
        count = 0;
      } else {
        max = Math.max(max, ++count);
      }
    }

    return max;
  }
}
